atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -279,6 +282,12 @@ LockWatchingCache lockWatchingCache() {
         thread.start();
     };
 
+    @Value.Check
+    protected void check() {
+        Preconditions.checkState(!(runtimeConfigSupplier().isPresent() && runtimeConfig().isPresent()),
+                "Either Refreshable or Supplier of runtime config must be provided or none, but not both");
```

Comment:
This maintains the current setup, where it's possible to not provide the runtime config (used for InMemory configuration).

Reply:
nit: probably enough to just say `Cannot provide both Refreshable and Supplier of runtime config`?

Reply:
Done